tests/Integration/PrestaShopBundle/Controller/Sell/Customer/Address/AddressControllerTest.php

```diff
@@ -0,0 +1,249 @@
+        $router = $client->getKernel()->getContainer()->get('router');
+        $addressDeleteUrl = $router->generate('admin_addresses_delete', ['addressId' => $this->testAddressId]);
+        $crawler = $client->request('POST', $addressDeleteUrl);
+        $addresses = $this->getAddressList($crawler);
+        self::assertEquals(2, count($addresses));
+    }
+
+    /**
+     * Tests filter for first name
+     */
+    public function testFirstNameFilter(): void
+    {
+        $client = static::createClient();
+        $client->followRedirects(true);
+        $router = $client->getKernel()->getContainer()->get('router');
+        $addressUrl = $router->generate('admin_addresses_index');
+        $crawler = $client->request('GET', $addressUrl);
+        $addresses = $this->getAddressList($crawler);
+        /* Make sure we have 3 addresses before we filter by firstname */
+        self::assertEquals(3, count($addresses));
+        $filterForm = $this->fillFiltersForm($crawler, ['address[firstname]' => $this->getTestAddress()->getFirstName()]);
+        $crawler = $client->submit($filterForm);
+        $addresses = $this->getAddressList($crawler);
+        self::assertEquals(1, count($addresses));
+        $this->validateTestAddressExists($addresses);
+    }
+
+    /**
+     * Tests filter for first name
+     */
+    public function testIdFilter(): void
+    {
+        $client = static::createClient();
+        $client->followRedirects(true);
+        $router = $client->getKernel()->getContainer()->get('router');
+        $addressUrl = $router->generate('admin_addresses_index');
+        $crawler = $client->request('GET', $addressUrl);
+        $addresses = $this->getAddressList($crawler);
+        /* Make sure we have 3 addresses before we filter by firstname */
+        self::assertEquals(3, count($addresses));
+        $filterForm = $this->fillFiltersForm($crawler, ['address[id_address]' => $this->getTestAddress()->getId()]);
+        $crawler = $client->submit($filterForm);
+        $addresses = $this->getAddressList($crawler);
+        self::assertEquals(1, count($addresses));
+        $this->validateTestAddressExists($addresses);
+    }
+
+    /**
+     * Validates that test address exists in provided list
+     *
+     * @param Address[]
+     */
+    private function validateTestAddressExists(array $addresses): void
+    {
+        $addressIds = [];
+
+        foreach ($addresses as $address) {
+            $addressIds[] = $address->getId();
+        }
+        self::assertTrue(in_array($this->getTestAddress()->getId(), $addressIds));
+    }
+
+    /**
+     * Creates test address that can be used for testing filters
+     *
+     * @param Client $client
+     */
+    private function createTestAddress(Client $client): void
+    {
+        $router = $client->getKernel()->getContainer()->get('router');
+        $createAddressUrl = $router->generate('admin_addresses_create');
+        $crawler = $client->request('GET', $createAddressUrl);
```

Comment:
I'm impressed we dont have to deal with authentication (but that's very useful!)

Reply:
Yeah when you figured out how DomCrawler works this makes them pretty straightforward, so that's really cool.